From a0d8577b09a1558763daaf031dc8186a47c4d2aa Mon Sep 17 00:00:00 2001
From: Jianpeng Chang <jianpeng.chang.cn@windriver.com>
Date: Mon, 3 Nov 2025 14:23:24 +0800
Subject: [PATCH] ARM: dts: enable cgroup memory controller by default

The current default boot arguments include cgroup_disable=memory. this
setting was introduced by commit 3a44d1c ("BCM2708: Add core Device Tree
support").

However, the Raspberry Pi 4 has sufficient memory to support the cgroup
memory controller, This functionality was supported in previous versions
without being explicitly disabled, so its current disabling appears to be
mistaken.

Remove cgroup_disable=memory from the boot arguments to leave the cgroup
memory controller enabled by default.

Upstream-Status: Submitted

Signed-off-by: Jianpeng Chang <jianpeng.chang.cn@windriver.com>
---
 arch/arm/boot/dts/broadcom/bcm270x.dtsi        | 2 +-
 arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi | 2 +-
 arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/broadcom/bcm270x.dtsi b/arch/arm/boot/dts/broadcom/bcm270x.dtsi
index 678bee7d96e7..187f95f93a20 100644
--- a/arch/arm/boot/dts/broadcom/bcm270x.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm270x.dtsi
@@ -4,7 +4,7 @@
 / {
 	chosen: chosen {
 		// Disable audio by default
-		bootargs = "coherent_pool=1M snd_bcm2835.enable_headphones=0 cgroup_disable=memory";
+		bootargs = "coherent_pool=1M snd_bcm2835.enable_headphones=0";
 		stdout-path = "serial0:115200n8";
 	};
 
diff --git a/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi b/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
index 4773b7bd2674..ba5f2f1494fb 100644
--- a/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
@@ -3,7 +3,7 @@
 
 / {
 	chosen {
-		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 cgroup_disable=memory numa_policy=interleave nvme.max_host_mem_size_mb=0";
+		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 numa_policy=interleave nvme.max_host_mem_size_mb=0";
 	};
 
 	__overrides__ {
diff --git a/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi b/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi
index c77e280ccd16..ba389fa3c43a 100644
--- a/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi
@@ -24,7 +24,7 @@ minibt: bluetooth {
 
 / {
 	chosen {
-		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 cgroup_disable=memory";
+		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0";
 	};
 
 	aliases {
-- 
2.51.2

